store/sqlstore/user_store.go

```diff
@@ -1536,3 +1536,87 @@ func (us SqlUserStore) GetUsersBatchForIndexing(startTime, endTime int64, limit
 		result.Data = usersForIndexing
 	})
 }
+
+func (us SqlUserStore) GetUsersPermittedToTeam(teamID string) store.StoreChannel {
+	return store.Do(func(result *store.StoreResult) {
+		query := us.usersQuery.
+			Where(`Id IN (
+				SELECT
+					GroupMembers.UserId
+				FROM
+					Teams
+					JOIN GroupTeams ON GroupTeams.TeamId = Teams.Id
+					JOIN UserGroups ON UserGroups.Id = GroupTeams.GroupId
+					JOIN GroupMembers ON GroupMembers.GroupId = UserGroups.Id
+				WHERE
+					Teams.Id = ?
+					AND Teams.GroupConstrained = TRUE
+					AND GroupTeams.DeleteAt = 0
+					AND UserGroups.DeleteAt = 0
+					AND Teams.DeleteAt = 0
```

Comment:
Consider removing `AND Teams.DeleteAt = 0` to allow this method to return meaningful results for deleted teams.